test/functional/sqlsrv/AEData.inc

```diff
@@ -1,113 +1,206 @@
+                             'This can create an implicit limit to the number of non-null varchar(max) or nvarchar(max) columns that can be created in a table.',
+                             'No special error is provided when the table is created (beyond the usual warning that the maximum row size exceeds the allowed maximum of 8060 bytes) or at the time of data insertion.',
+                             'This large row size can cause errors (such as error 512) during some normal operations, such as a clustered index key update, or sorts of the full column set, which users cannot anticipate until performing an operation.' );
+// Array containing all SQLSRV_SQLTYPE_ types to pass into a bind param prepare statement      
+                
+$sqlTypes = array(
+    'SQLSRV_SQLTYPE_BIGINT',
+    'SQLSRV_SQLTYPE_BINARY',
+    'SQLSRV_SQLTYPE_BIT',
+    'SQLSRV_SQLTYPE_CHAR',
+    'SQLSRV_SQLTYPE_DATE',
+    'SQLSRV_SQLTYPE_DATETIME',
+    'SQLSRV_SQLTYPE_DATETIME2',
+    'SQLSRV_SQLTYPE_DATETIMEOFFSET',
+    'SQLSRV_SQLTYPE_DECIMAL',
+    'SQLSRV_SQLTYPE_FLOAT',
+    'SQLSRV_SQLTYPE_IMAGE',
+    'SQLSRV_SQLTYPE_INT',
+    'SQLSRV_SQLTYPE_MONEY',
+    'SQLSRV_SQLTYPE_NCHAR',
+    'SQLSRV_SQLTYPE_NUMERIC',
+    'SQLSRV_SQLTYPE_NVARCHAR',
+    'SQLSRV_SQLTYPE_NTEXT',
+    'SQLSRV_SQLTYPE_REAL',
+    'SQLSRV_SQLTYPE_SMALLDATETIME',
+    'SQLSRV_SQLTYPE_SMALLINT',
+    'SQLSRV_SQLTYPE_SMALLMONEY',
+    'SQLSRV_SQLTYPE_TEXT',
+    'SQLSRV_SQLTYPE_TIME',
+    'SQLSRV_SQLTYPE_TIMESTAMP',
+    'SQLSRV_SQLTYPE_TINYINT',
+    'SQLSRV_SQLTYPE_UNIQUEIDENTIFIER',
+    'SQLSRV_SQLTYPE_VARBINARY',
+    'SQLSRV_SQLTYPE_VARCHAR',
+    'SQLSRV_SQLTYPE_XML'
+);      
+// Checks if the current error is the incompatible types error
+// if so, state which sql type is incompatible with which data type
+function is_incompatible_types_error( $dataType, $sqlType )
+{
+    $errors = sqlsrv_errors();
+    foreach ( $errors as $error )
+    {
+        // 22018 is the SQLSTATE for the operand crash error for incompatible types
+        if ( $error['SQLSTATE'] == 22018 )
+        {
+            echo "Encrypted $sqlType is incompatible with encrypted $dataType\n";
+        }
+    }
+}
+function get_default_size_prec( $sqlType )
+{
+    if ( $sqlType == 'SQLSRV_SQLTYPE_DECIMAL' )
+        $sqlType .= "(18, 5)";
+    elseif ( $sqlType == 'SQLSRV_SQLTYPE_NUMERIC' )
+        $sqlType .= "(10, 5)";
+    elseif ( $sqlType == 'SQLSRV_SQLTYPE_CHAR' || $sqlType == 'SQLSRV_SQLTYPE_NCHAR' )
+        $sqlType .= "(5)";
+    return $sqlType;
+}
+
+// get sqlType constant value from string
+function get_sqlType_constant( $sqlType )
+{
+    switch ( $sqlType ) {
+		case 'SQLSRV_SQLTYPE_BIGINT':
+		    return SQLSRV_SQLTYPE_BIGINT;
+			break;
+		case 'SQLSRV_SQLTYPE_BINARY':
+		    return SQLSRV_SQLTYPE_BINARY;
+			break;
+		case 'SQLSRV_SQLTYPE_BIT':
+		    return SQLSRV_SQLTYPE_BIT;

```

Comment:
for getting the constant from a string, you can do `constant('SQLSRV_SQLTYPE_BIT')`
for getting the constant of sqltypes with parenthesis, you can do `call_user_func('SQLSRV_SQLTYPE_CHAR', 5)`